E2E: "CSI attach test using HostPath driver" fails

The "CSI attach test using HostPath driver" test from `kubernetes/test/e2e/storage/csi_volumes.go` fails for me:
```
$ go run hack/e2e.go -- --provider=local --test --test_args="--ginkgo.focus=CSI.attach.test.using.HostPath.driver"
....
• Failure [40.263 seconds]
[sig-storage] [Serial] CSI Volumes
/nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/utils/framework.go:22
  CSI attach test using HostPath driver [Serial][Feature:CSISkipAttach]
  /nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/csi_volumes.go:125
    attachable volume needs VolumeAttachment [It]
    /nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/csi_volumes.go:165

    Failed to create CSIDriver: CSIDriver.csi.storage.k8s.io "csi-hostpath" is invalid: []: Invalid value: map[string]interface {}{"kind":"CSIDriver", "apiVersion":"csi.storage.k8s.io/v1alpha1", "metadata":map[string]interface {}{"creationTimestamp":"2018-09-27T12:48:19Z", "generation":1, "uid":"9c1caf02-c253-11e8-a335-fcaa1497a416", "name":"csi-hostpath"}, "spec":map[string]interface {}{"attachRequired":true, "podInfoOnMountVersion":interface {}(nil)}}: validation failure list:
    spec.podInfoOnMountVersion in body must be of type string: "null"
    Expected error:
        <*errors.StatusError | 0xc422404900>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
                Status: "Failure",
                Message: "CSIDriver.csi.storage.k8s.io \"csi-hostpath\" is invalid: []: Invalid value: map[string]interface {}{\"kind\":\"CSIDriver\", \"apiVersion\":\"csi.storage.k8s.io/v1alpha1\", \"metadata\":map[string]interface {}{\"creationTimestamp\":\"2018-09-27T12:48:19Z\", \"generation\":1, \"uid\":\"9c1caf02-c253-11e8-a335-fcaa1497a416\", \"name\":\"csi-hostpath\"}, \"spec\":map[string]interface {}{\"attachRequired\":true, \"podInfoOnMountVersion\":interface {}(nil)}}: validation failure list:\nspec.podInfoOnMountVersion in body must be of type string: \"null\"",
                Reason: "Invalid",
                Details: {
                    Name: "csi-hostpath",
                    Group: "csi.storage.k8s.io",
                    Kind: "CSIDriver",
                    UID: "",
                    Causes: [
                        {
                            Type: "FieldValueInvalid",
                            Message: "Invalid value: map[string]interface {}{\"apiVersion\":\"csi.storage.k8s.io/v1alpha1\", \"metadata\":map[string]interface {}{\"creationTimestamp\":\"2018-09-27T12:48:19Z\", \"generation\":1, \"uid\":\"9c1caf02-c253-11e8-a335-fcaa1497a416\", \"name\":\"csi-hostpath\"}, \"spec\":map[string]interface {}{\"attachRequired\":true, \"podInfoOnMountVersion\":interface {}(nil)}, \"kind\":\"CSIDriver\"}: validation failure list:\nspec.podInfoOnMountVersion in body must be of type string: \"null\"",
                            Field: "[]",
                        },
                    ],
                    RetryAfterSeconds: 0,
                },
                Code: 422,
            },
        }
        CSIDriver.csi.storage.k8s.io "csi-hostpath" is invalid: []: Invalid value: map[string]interface {}{"kind":"CSIDriver", "apiVersion":"csi.storage.k8s.io/v1alpha1", "metadata":map[string]interface {}{"creationTimestamp":"2018-09-27T12:48:19Z", "generation":1, "uid":"9c1caf02-c253-11e8-a335-fcaa1497a416", "name":"csi-hostpath"}, "spec":map[string]interface {}{"attachRequired":true, "podInfoOnMountVersion":interface {}(nil)}}: validation failure list:
        spec.podInfoOnMountVersion in body must be of type string: "null"
    not to have occurred

    /nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/csi_volumes.go:227
------------------------------
SSSSSSSSSSSSSSSSSSep 27 14:48:37.353: INFO: Running AfterSuite actions on all node
Sep 27 14:48:37.353: INFO: Running AfterSuite actions on node 1


Summarizing 2 Failures:

[Fail] [sig-storage] [Serial] CSI Volumes CSI attach test using HostPath driver [Serial][Feature:CSISkipAttach] [It] non-attachable volume does not need VolumeAttachment 
/nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/csi_volumes.go:227

[Fail] [sig-storage] [Serial] CSI Volumes CSI attach test using HostPath driver [Serial][Feature:CSISkipAttach] [It] attachable volume needs VolumeAttachment 
/nvme/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/csi_volumes.go:227

Ran 3 of 1813 Specs in 158.815 seconds
FAIL! -- 1 Passed | 2 Failed | 0 Pending | 1810 Skipped --- FAIL: TestE2E (158.90s)
FAIL
```

I ran this after building Kubernetes from source (some recent master revision) and starting the cluster with:
```
RUNTIME_CONFIG= ALLOW_PRIVILEGED=1 FEATURE_GATES="BlockVolume=true,MountPropagation=true,KubeletPluginsWatcher=true,CSINodeInfo=true,CSIDriverRegistry=true"  hack/local-up-cluster.sh -O
```

--- a/staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go
+++ b/staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go
@@ -63,7 +63,7 @@ type CSIDriverSpec struct {
 	// If value is not specified, default is false -- meaning attach will not be
 	// called.
 	// +optional
-	AttachRequired *bool `json:"attachRequired"`
+	AttachRequired *bool `json:"attachRequired,omitempty"`
 
 	// If specified, podInfoRequiredOnMount indicates this CSI volume driver
 	// requires additional pod information (like podName, podUID, etc.) during
@@ -77,7 +77,7 @@ type CSIDriverSpec struct {
 	// "csi.storage.k8s.io/pod.namespace": pod.Namespace
 	// "csi.storage.k8s.io/pod.uid": string(pod.UID)
 	// +optional
-	PodInfoOnMountVersion *string `json:"podInfoOnMountVersion"`
+	PodInfoOnMountVersion *string `json:"podInfoOnMountVersion,omitempty"`
 }
 
 // +genclient

